src/legacy/core_plugins/kibana/public/visualize/listing/visualize_listing.js

```diff
@@ -94,15 +94,18 @@ export function VisualizeListingController($injector, createNewVis) {
       });
   };
 
-  this.deleteSelectedItems = function deleteSelectedItems(selectedIds) {
-    return visualizationService.delete(selectedIds)
-      .catch(error => {
-        toastNotifications.addError(error, {
-          title: i18n.translate('kbn.visualize.visualizeListingDeleteErrorTitle', {
-            defaultMessage: 'Error deleting visualization',
-          }),
-        });
+  this.deleteSelectedItems = function deleteSelectedItems(_ids, selectedItems) {
+    return Promise.all(
+      selectedItems.map(item => {
+        return savedObjectClient.delete(item.savedObjectType, item.id);
```

Comment:
The "old" delete function also called `chrome.untrackNavLinksForDeletedSavedObjects` in `src/legacy/ui/public/saved_objects/saved_object_loader.js` line 71, we probably also have to do that, even though I'm not sure what is does.

Reply:
Interesting. Good catch. Here's the comment for that function:
`Clear last url for deleted saved objects to avoid loading pages with "Could not locate..."`

Sounds like something we need.